Sprint_6 .

ManevichRR opened the pull request:
JaCoCo показывает что Feline не покрыт тестами, но вроде как покрыт
Не могу понять в чем проблема 

- gitignore edited to exclude extra Manrojw 
- add jacoco report after Java switched to 11v Manrojw 
- Changed Java to 11 Manrojw 
- add jacoco report Manrojw 
- Finished FelineTest Edited dependencies for JaCoCo Manrojw 
- LionTest done. Edited constructor for Lion class Manrojw T
- CatTest done. Manrojw Today 2:43 PM
- Переписал без использования org.mockito.MockitoAnnotations; Тк мы его не использовали в обучении Manrojw 
- Runner for Feline Manrojw 
- Test for Feline Manrojw 
- Editing cz texted wrong comment for last commit. Right comment for prev commit is: 'DI done' Manrojw 
- Added Test classes Manrojw 
- Added Test classes Manrojw 
- added dependencies Mockito JUnit4 and Manrojw 
- added dependencies Mockito JUnit4 and Manrojw 
- test with ssh Manrojw 
- Test Manrojw 
- Initial commit ManevichRR*

Commit:
added dependencies
Mockito JUnit4 and

Commit:
added dependencies
Mockito JUnit4 and

Commit:
Added Test classes

Commit:
Added Test classes

Commit:
Editing cz texted wrong comment for last commit. Right comment for prev commit is: 'DI done'

Commit:
Test for Feline

Commit:
Runner for Feline

Commit:
Переписал без использования org.mockito.MockitoAnnotations; Тк мы его не использовали в обучении

Commit:
CatTest done.

Commit:
LionTest done.
Edited constructor for Lion class

Commit:
Finished FelineTest
Edited dependencies for JaCoCo

Commit:
add jacoco report

Commit:
Changed Java to 11

Commit:
add jacoco report after Java switched to 11v

Commit:
gitignore edited to exclude extra

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,47 @@
+        List<String> actualFood = lion.getFood();
+        assertEquals(expectedFood, actualFood);
+    }
+}
```
Не хватает теста для Самки ( лучше сделать с параметризацией) и теста на исключение. 
нужно добавить

digranate replied:
Не увидела теста на исключение ( вызов конструктора Lion с невалидным параметром)
Исключение тестируем через метод assertThrows или параметр в аннотации Test.
Нужно добавить

pomegranate282922 reviewed (commented):
Работа хорошая! Использованы моки и стабы, сделана изоляция. 
есть замечания:
1. В тесте на feline нужно убрать моки 
2. Добавить недостающие тесты 
3. Остальное по мелочи

Commit:
Чутка с китенами не понял, поясни плиз
Пока все в кашу, к сожалению

Commit:
Доработки доделки

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @Test
+    public void testGetKittens() throws Exception {
```
Здесь нехорошо то, что эти тесты повторятся дважды, так как лежат в параметризованном классе. Их нужно разделить.

pomegranate282922 reviewed (commented):
Многое исправил, но два замечания есть -
1. Задвоились тесты 
2. Нет теста на исключение

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -40,9 +42,12 @@ public void testDoesHaveMane() throws Exception {
+    }
     @Test
-    public void testGetKittens() throws Exception {
+    public void testGetKittens() throws Exception { //Че-то я тут сломался, не понимаю что сделать сделал Бифор :(
```
Этот тест и другие повторятся при запуске дважды, так как находятся в параметризованном классе, но сами не используют параметр. Поэтому нужно их вынести в другой класс.

pomegranate282922 reviewed (commented):
Не исправлены замечания:
1. Написала пояснение, про параметризованный класс 
2. Нет теста на выпадение исключения при вызове конструктора Lion с невалидным параметром

Commit:
Класс вынес
Невалидный сделал
Вроде подразобрался, но все равно не супер уверен в знаниях (

pomegranate282922 reviewed (approved):
Да, все так! Зачет!
